lib/dalli/client.rb

```diff
@@ -45,10 +45,18 @@ class Client
     #                   useful for injecting a FIPS compliant hash object.
     # - :protocol - one of either :binary or :meta, defaulting to :binary.  This sets the protocol that Dalli uses
     #               to communicate with memcached.
+    # - :safe_get - if true Dalli will use getk to get key with value and check that the keys match. This protects against 
+    #               socket corruption at the cost of additional overhead to retrieve and compare keys. 
+    #               This is currently only implemented for the binary protocol. Default: false
     #
     def initialize(servers = nil, options = {})
       @normalized_servers = ::Dalli::ServersArgNormalizer.normalize_servers(servers)
       @options = normalize_options(options)
+
+      if @options[:safe_get].present? && @options[:safe_get] == true && @options[:protocol].present? && @options[:protocol] != :binary
```

Comment:
In case you want to share this to upstream: `present?` is an ActiveSupport method and this gem doesn't depend on it.

Reply:
Good catch thanks. I assume !`nil?` is the only alternative?

Reply:
https://github.com/cornu-ammonis/dalli/pull/2/commits/118cf5ac123cf5e86624bd2171d4216cba709531